### PR TITLE
Misc updates for v20

### PR DESCRIPTION
**Pull request by emveepee**
1. Provide an option to sort recordings into their respective named directory order.  
2. Add a fast poll for the server on startup

### CODE REVIEW

File: src/Recordings.cpp [hunk 1]
```diff
@@ -204,7 +218,12 @@ bool Recordings::UpdatePvrRecording(const tinyxml2::XMLNode* pRecordingNode, kod
   tag.SetTitle(title);
 
   XMLUtils::GetString(pRecordingNode, "start_time_ticks", buffer);
-  tag.SetRecordingTime(stol(buffer));
+  try {
+    tag.SetRecordingTime(stol(buffer));
```
Comment (phunkyfish) [inline src/Recordings.cpp]: Qualify `std` calls. So `std::stol()`

File: src/Recordings.cpp [hunk 2]
```diff
@@ -358,6 +377,22 @@ bool Recordings::UpdatePvrRecording(const tinyxml2::XMLNode* pRecordingNode, kod
       }
     }
   }
+  if (m_settings.m_showRoot)
+  {
+    const std::string original = tag.GetDirectory();
+    std::string root = "Default";
```
Comment (phunkyfish) [inline src/Recordings.cpp]: Constant instead of string literal?

**Review by phunkyfish — CHANGES_REQUESTED** [review]

File: src/Recordings.cpp (hunk 1 above)
Comment (emveepee) [reply src/Recordings.cpp]: Fixed

File: src/Recordings.cpp (hunk 2 above)
Comment (emveepee) [reply src/Recordings.cpp]: I don't know what you mean so I removed const

Comment (phunkyfish) [reply src/Recordings.cpp]: Apologies. I meant the the line with the “Default”.

The const was perfectly fine where it was and correct.

**Review by phunkyfish — APPROVED** [review]

File: src/Recordings.cpp (hunk 2 above)
Comment (emveepee) [reply src/Recordings.cpp]: @phunkyfish std::string root  changes and can't be a constant

Comment (phunkyfish) [reply src/Recordings.cpp]: Oh, right, my mistake.

**Comment by emveepee** [comment]
@phunkyfish did you get the review request?

File: src/pvrclient-nextpvr.cpp [hunk 3]
```diff
@@ -194,7 +199,10 @@ ADDON_STATUS cPVRClientNextPVR::Connect(bool sendWOL)
         SetConnectionState("Connnecting", PVR_CONNECTION_STATE_CONNECTING);
       }
       m_connectionState = PVR_CONNECTION_STATE_SERVER_UNREACHABLE;
-      m_nextServerCheck = time(nullptr) + 60;
+      if (time(nullptr) > m_firstSessionInitiate + 65 )
+        m_nextServerCheck = time(nullptr) + SLOW_CONNECT_POLL;
```
Comment (phunkyfish) [inline src/pvrclient-nextpvr.cpp:208]: `65` should likely be defined also.

File: src/pvrclient-nextpvr.h [hunk 4]
```diff
@@ -142,15 +138,13 @@ class ATTR_DLL_LOCAL cPVRClientNextPVR : public kodi::addon::CInstancePVRClient
 private:
   void ConfigurePostConnectionOptions();
   const CNextPVRAddon& m_base;
-  void Close();
+
```
Comment (phunkyfish) [inline src/pvrclient-nextpvr.h:141]: 1 empty line should be enough

**Review by phunkyfish — CHANGES_REQUESTED** [review]

**Review by phunkyfish — APPROVED** [review]

**Comment by emveepee** [comment]
@phunkyfish thanks, this time you even beat me to github.  NP with the delay I have been testing the change with CoreElec and LibreElec while I have waited.

**Pull request merged**